TypeSystem/flow-analysis/reachability_switch_expression_A04_t03.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion switch expression: If `N` is a switch expression of the form
+/// `switch (E) {alternatives}` then:
+/// - Let `before(E) = split(before(N))`.
+/// - For each case `Cj` in `alternatives` with expression `Ej`, for `j` in
+///   `1 .. k`:
+///   - Let `before(Ej) = after(E)`.
+/// - Let `after(N) = join(after(E1), after(E2), .. after(Ek))`.
+/// TODO (sgrekhov): there is no switch expression in the flow analysis spec yet
+/// (April 2025). Update the assertion after spec update.
+///
+/// @description Checks that if a type `T` is made a type of interest in
+/// `alternatives` then it cannot be promoted in other `alternatives`.
```

Comment:
How can this be phrased? It is not possible for one alternative to _prevent_ promotion of a variable `s` in other alternatives unless it is because it makes `s` non-promotable entirely (`() { s = something; })`). What we _can_ say is that making `T` a type of interest for a given variable in one alternative does not make `T` a type of interest for that variable in any other cases.

Perhaps:
```suggestion
/// @description Checks that if a type `T` is made a type of interest for `s`
/// in an alternative then there is no guarantee that `T` is of interest for `s`
/// in any other alternative.
```

Reply:
Thank you. Updated.